### PR TITLE
 18313: CT: Add ability to search using state abbreviation

### PR DESCRIPTION
**Pull request by elder-joseph**
## Description
As a Comparison Tool user, I need to be able to search by state abbreviation, so I can find schools in my state besides ones that have the state in the name.

[Originating Issue](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/18313)

This is the back end PR to update search results when the `gibct_state_search` feature flag is passed.

## Testing done
Testing passes locally, additional unit tests added.

## Screenshots
N/A

## Acceptance criteria
For the non VETTEC search:
- [x] 1. Search by state abbreviation (ex. TX, AL, SC) returns all institutions within that state.
- [x] 2. Search by "city, state abbreviation" returns all institutions within that city and state.

For the VETTEC search:
- [x] 3. Search by state abbreviation (ex. TX, AL, SC) returns all institutions within that state.
- [x] 4. Search by "city, state abbreviation" returns all institutions within that city and state.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by sonntag-adam — APPROVED** [review]

File: Gemfile [hunk 1]
```diff
@@ -97,6 +97,7 @@ group :development, :test do
   gem 'faker', '~> 1.6', '>= 1.6.2'
   gem 'simplecov'
   gem 'vcr', '~> 3.0', '>= 3.0.1'
+  gem 'vets_json_schema', git: 'https://github.com/department-of-veterans-affairs/vets-json-schema', branch: 'master'
```
Comment (LindseySaari) [inline Gemfile:100]: I think if you're introducing a new gem, this should be broken out into a separate PR

File: .rubocop.yml [hunk 2]
```diff
@@ -107,6 +107,7 @@ Metrics/ClassLength:
     - "app/controllers/groups_controller.rb"
     - "app/controllers/uploads_controller.rb"
     - "app/models/upload.rb"
+    - "app/controllers/v0/institution_programs_controller.rb"
```
Comment (LindseySaari) [inline .rubocop.yml:110]: Is there a way to break up the method instead of disabling this rubopcop check?

File: Gemfile (hunk 1 above)
Comment (elder-joseph) [reply Gemfile:100]: I can go ahead and break this out into a new PR.

File: .rubocop.yml [hunk 3]
```diff
@@ -96,7 +96,7 @@ Metrics/MethodLength:
     - "app/controllers/v0/institutions_controller.rb"
     - "lib/roo_helper/loader.rb"
     - "lib/seed_utils.rb"
-
+    - "app/controllers/v0/institution_programs_controller.rb"
```
Comment (LindseySaari) [inline .rubocop.yml:99]: Is there a way to break up the method instead of disabling this rubopcop check?

Comment (elder-joseph) [reply .rubocop.yml]: Hi @LindseySaari, I just wanted to follow up. I was able to merge in the separate Gemfile specific PR yesterday and was also able to break up this method which resolved any rubocop/code climate issues for this PR.

**Review by LindseySaari — APPROVED** [review]
